### PR TITLE
Misc infra changes

### PR DESCRIPTION
**Pull request by jareware**
Bunch of tiny things I've been dragging my feet pushing up.

### CODE REVIEW

File: infra/modules/main/backend.tf [hunk 1]
```diff
@@ -7,7 +7,20 @@ resource "aws_s3_bucket" "backend_code" {
     target_bucket = var.s3_logs_bucket
     target_prefix = "${var.name_prefix}-backend-code/"
   }
+}
 
+locals {
```
Comment (cutepig) [inline infra/modules/main/backend.tf:12]: I guess this is good, removing duplication of environment variable declarations 👌

File: infra/modules/aws_lambda_api/certificate.tf [hunk 2]
```diff
@@ -3,8 +3,8 @@
 resource "aws_acm_certificate" "this" {
   provider          = aws.us_east_1 # because ACM is only available in the "us-east-1" region
   domain_name       = var.api_domain
-  validation_method = "DNS" # the required records are created below
-  tags              = var.tags
+  validation_method = "DNS"                                       # the required records are created below
+  tags              = merge(var.tags, { Name = var.name_prefix }) # NOTE: the "Name" tag is special in ACM, in that it appears in web console listings (this is just for convenience, though)
```
Comment (cutepig) [inline infra/modules/aws_lambda_api/certificate.tf:7]: Good comment 🔥

File: infra/modules/main/cloudwatch.tf [hunk 3]
```diff
@@ -45,6 +45,18 @@ resource "aws_cloudwatch_dashboard" "overview" {
                         {
                             "label": "Second result articles (https://www.hs.fi/kotimaa/art-2000006463218.html & https://www.is.fi/kotimaa/art-2000006463005.html)",
                             "value": "2020-04-03T16:00:00+03:00"
+                        },
```
Comment (cutepig) [inline infra/modules/main/cloudwatch.tf:48]: Now what is this dashboard?

File: infra/modules/main/backend.tf (hunk 1 above)
Comment (jareware) [reply infra/modules/main/backend.tf:12]: Yeah, that was the idea! So we're less likely to update API env but forget making the corresponding change to worker env, or vice versa.

File: infra/modules/main/cloudwatch.tf (hunk 3 above)
Comment (jareware) [reply infra/modules/main/cloudwatch.tf:48]: You can enjoy it [here in our AWS console](https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#dashboards:)!

**Pull request merged**